mrc/control/movement_prediction/utils.py

```diff
@@ -0,0 +1,89 @@
+import numpy as np
+
+
+def _polar2cartesian(vector):
+    r, theta = vector
+    return np.asarray([r * np.cos(theta), r * np.sin(theta)])
+
+
+def _cartesian2polar(vector):
+    x, y = vector
+    return np.asarray([np.sqrt(x ** 2 + y ** 2), np.arctan2(y, x)])
+
+
+def _normalize_angle(angle):
+    normalized = np.pi / 2 - angle
+    if normalized <= -np.pi:
+        normalized += 2 * np.pi
+    elif normalized > np.pi:
+        normalized -= 2 * np.pi
+    return normalized
+
+
+def _normalize_polars(coords):
+    return coords[0], _normalize_angle(coords[1])
+
+
+def sum_vectors(*vecs):
+    norm_vecs_c = [_polar2cartesian(_normalize_polars(vector)) for vector in vecs]
+    vec_res = np.sum(norm_vecs_c)
+    return _normalize_polars(_cartesian2polar(vec_res))
+
+
+def translate(v_old, distance, rotation):
+    """Function recalculating position of old vector in new coordinate system
+
+    Parameters
+    ----------
+    v_old : array-like
+        Vector in old coordinate system. It should have two fields, first one being radius, second being angle.
+    distance : float
+        Length of vector that coordinate system was moved along.
+    rotation : float
+        Angle of coordinate system's rotation in radians.
+
+    Returns
+    -------
+    ndarray
+        Input vector in new coordinate system.
+
+    Notes
+    -----
+        Angle should be measured clockwise and having value of 0 along vertical coordinate system axis.
+    """
+    r_old, theta_old = v_old
+    theta_new = _normalize_angle(theta_old) + rotation if r_old != 0 else 0
```

Comment:
What if theta_new is not in [-PI, PI]? It would be nice to split _normalize_angle into two functions: 1. change coordinate system 2. make angle lie in [-PI, PI] interval.